zio-http/src/main/scala/zhttp/http/Http.scala

```diff
@@ -512,6 +521,41 @@ object Http {
    */
   def html(view: Html): HttpApp[Any, Nothing] = Http.response(Response.html(view))
 
+  /**
+   * Creates an HTTP app to serve static resource files from a local directory.
+   */
+  def fromPath(root: jPath): HttpApp[Any, Nothing] = {
+    def contentTypeHeader(file: java.io.File): Header   =
+      (HttpHeaderNames.CONTENT_TYPE, Files.probeContentType(file.toPath))
+    def contentLengthHeader(file: java.io.File): Header = (HttpHeaderNames.CONTENT_LENGTH, file.length().toString)
+
+    def responseHttp(file: File) = responseZIO {
+      for {
+        data <- Task(HttpData.fromFile(file))
+        res  <- ZIO.succeed(Response(data = data))
+      } yield res
+    } @@ updateHeaders(_.addHeaders(Headers(Chunk(contentTypeHeader(file), contentLengthHeader(file)))))
```

Comment:
Middleware isn't required. We can add the headers directly while creating the response.